src/main/java/net/staticsnow/nexus/repository/apt/internal/FacetHelper.java

```diff
@@ -15,84 +15,182 @@
-	public static Content toContent(final Asset asset, final Blob blob) {
-		final Content content = new Content(new BlobPayload(blob, asset.requireContentType()));
-		Content.extractFromAsset(asset, hashAlgorithms, content.getAttributes());
-		return content;
-	}
+public class FacetHelper
+{
+  public static final List<HashAlgorithm> hashAlgorithms = Arrays.asList(MD5, SHA1, SHA256);
+
+  private FacetHelper() {
+    // empty
+  }
+
+  public static List<ContentSpecifier> getReleaseIndexSpecifiers(AptFacet facet) {
+    if (facet.isFlat()) {
+      return Arrays.asList(
+          new ContentSpecifier("Release", SnapshotItem.Role.RELEASE_INDEX),
+          new ContentSpecifier("Release.gpg", SnapshotItem.Role.RELEASE_SIG),
+          new ContentSpecifier("InRelease", SnapshotItem.Role.RELEASE_INLINE_INDEX)
+      );
+    }
+    else {
+      String dist = facet.getDistribution();
+      return Arrays.asList(
+          new ContentSpecifier("dists/" + dist + "/Release", SnapshotItem.Role.RELEASE_INDEX),
+          new ContentSpecifier("dists/" + dist + "/Release.gpg", SnapshotItem.Role.RELEASE_SIG),
+          new ContentSpecifier("dists/" + dist + "/InRelease", SnapshotItem.Role.RELEASE_INLINE_INDEX)
+      );
+    }
+
+  }
+
+  public static List<ContentSpecifier> getReleasePackageIndexes(AptFacet facet, String component, String arch) {
+    if (facet.isFlat()) {
+      return Arrays.asList(
+          new ContentSpecifier("Packages", SnapshotItem.Role.PACKAGE_INDEX_RAW),
+          new ContentSpecifier("Packages.gz", SnapshotItem.Role.PACKAGE_INDEX_GZ),
+          new ContentSpecifier("Packages.bz2", SnapshotItem.Role.PACKAGE_INDEX_BZ2)
+      );
+    }
+    else {
+      String dist = facet.getDistribution();
+      return Arrays.asList(
+          new ContentSpecifier("dists/" + dist + "/" + component + "/binary-" + arch + "/Packages",
+              SnapshotItem.Role.PACKAGE_INDEX_RAW),
+          new ContentSpecifier("dists/" + dist + "/" + component + "/binary-" + arch + "/Packages.gz",
+              SnapshotItem.Role.PACKAGE_INDEX_GZ),
+          new ContentSpecifier("dists/" + dist + "/" + component + "/binary-" + arch + "/Packages.bz2",
+              SnapshotItem.Role.PACKAGE_INDEX_BZ2)
+      );
+    }
+
+  }
+
+  public static String determineContentType(SnapshotItem item) {
+    String ct = item.content.getContentType();
+    if (ct == null) {
+      switch (item.specifier.role) {
+        case RELEASE_INDEX:
+        case RELEASE_INLINE_INDEX:
+        case PACKAGE_INDEX_RAW:
+          return AptMimeTypes.TEXT;
+        case RELEASE_SIG:
+          return AptMimeTypes.SIGNATURE;
+        case PACKAGE_INDEX_BZ2:
+          return AptMimeTypes.BZIP;
+        case PACKAGE_INDEX_GZ:
+          return AptMimeTypes.GZIP;
+      }
+    }
+    return ct;
+  }
+
+  /**
+   * Find a component by it's name
+   *
+   * @return found component of null if not found
+   */
+  @Nullable
+  public static Component findComponent(final StorageTx tx,
```

Comment:
I added these and a few other methods as you'll see below, we do stuff like this internally just to clean up things a bit.